pkg/diff/diff.go

```diff
@@ -68,9 +122,10 @@ type Syncer struct {
 	processor     crud.Registry
 	postProcessor crud.Registry
 
-	eventChan chan crud.Event
-	errChan   chan error
-	stopChan  chan struct{}
+	eventChan  chan crud.Event
+	errChan    chan error
+	stopChan   chan struct{}
+	ResultChan chan EntityAction
```

Comment:
I just noticed that this field is the only one that would become exported.

While this would work I'd say that it'd be better to return this channel via a method. Even better (but probably out of scope for this PR) introduce a subscription mechanism where each interested party would get its own notification channel.

For now I believe just returning a channel via a method would do so that users do not need to mess with `Syncer`'s fields.

Reply:
Method's fine, though I doubt we're likely to refactor around multiple Syncer subscribers anytime soon. We don't even persist the things across update operations, and the design til now has essentially been "library clients create and are the exclusive owners of their Syncers". Offhand I don't know of any use cases where we'd want to refactor it for multiple clients and/or persistence, but that'd be the driving factor for such a change should it arise.